### ISSUE TITLE
docker-compose exec work on "run" containers

### ISSUE THREAD

**Issue by andyneff** [docker/compose#6549]
## Example use case

`docker-compose.yml`

```
version: "3.2"
services:
  foo:
    image: debian:9
    command: sleep 10000
```

1. `docker-compose run -d foo`
2. `docker-compose exec foo bash`

### Current Result:

```
ERROR: No container found for foo_1
```

### Desired result

It would be nice if not only did `docker-compose exec` check for containers created by `up`, but also `run`. This is already done for commands like `down` and `ps`

**Example:** `docker-compose ps`

```
          Name              Command     State   Ports
-----------------------------------------------------
foo_foo_run_382b47cedaf4   sleep 10000  Up 
```

### Alternative solutions

If for some reason mixing the `up`ed and `run`ed containers together in `exec` is incompatible with use cases/maintaining backwards compatibility, then specifying a flag that either "Only execs using `run` containers" or combines the list of `up` and `exec` containers. Which ever you think is best if this is needed.

### Current work around

I currently have to have to use a mix of `bash`, `docker-compose ps`, or `docker ps` commands to find the container I want to exec into

**Comment by collin5** [comment]
This would be a good one to have. Will be giving this a try perhaps this weekend.

**Comment by collin5** [comment]
not stale.

**Comment by glours** [comment]
@collin5 could you fix the failed test in your PR #6647 ?

**Comment by glours** [comment]
thanks 🙂

**Comment by ndeloof** [comment]
How would you identify the container to exec into if you had ran `docker-compose run` twice ?

**Comment by andyneff** [comment]
## TL; DR

"Just always use the **last** one" would be a good starting point

---

I personally do this in a script (We'll call this Idea 1):

- If there is only one -> exec into that one
- If there are more than one:
    - List them, number from 1 to X
    - Wait for you to select which one you want.

Of course, this might not be how we want to possibly strategies:

- Idea 2: Just always use the **last** one
    - Maybe you add a argument, say `--index` or something. Of course that index would be blindly picking from the list from idea 1
- Idea 3: Use Idea 1 if you have a TTY, else use Idea 2. I'm not talking about if the container has a tty or not, I mean if the docker-compose executable has an open tty or not, this is the same check that is often used to check if colors should be output or not, like by `grep` and other commands.

**Comment by collin5** [comment]
Hey @glours, sorry for the delay on this. Been caught up with some stuff at work lately. I've updated PR  #6647 and all tests should now be passing.

**Comment by kingbuzzman** [comment]
BAD BOT!

**Comment by nicolabeghin** [comment]
following

**Comment by turboasm** [comment]
we have the same issue, could we add some additional flag like --no-index in a future?

**Comment by Holist** [comment]
Is there a workaround for this or another Issue opened ?
docker-compose exec only accept service name (which dont work...)

**Comment by mshiltonj** [comment]
I am still having this problem with version 1.29.2. Would be nice if "It Just Works" ™

This initial report of this bug said the command returned an error message in this case: "ERROR: No container found for foo_1". I do not get any error text at all. It simply returns with no action or message.  It does return with an error code of 1. That's the only way I know that something went wrong.

My naive  workaround is a small bash function.

```
function de(){
   CONTAINER=$1
   shift
   echo "docker exec -it `docker ps | grep $CONTAINER | tail -n 1 | tr -s ' ' | awk '{ print $NF }'` $@"
   docker exec -it `docker ps | grep $CONTAINER | tail -n 1 | tr -s ' ' | awk '{ print $NF }'` $@
}
```

I run it with `de app bash` mostly. It grabs the last entry in `docker ps` that matches the first param, the passes the rest of the params to the docker exec -it CONTAINER command.
